src/mlpack/core/data/load_impl.hpp

```diff
@@ -370,10 +370,180 @@ bool Load(const std::string& filename,
 }
 
 // Load with mappings.  Unfortunately we have to implement this ourselves.
-template<typename eT, typename MapperType>
+template<typename eT, typename PolicyType>
 bool Load(const std::string& filename,
           arma::Mat<eT>& matrix,
-          MapperType& info,
+          DatasetMapper<PolicyType>& info,
+          PolicyType& policy,
```

Comment:
I think we should reduce duplicate codes, there are two solutions

1 : Use 

```
template<typename eT, typename PolicyType>
bool Load(const std::string& filename,
                 arma::Mat<eT>& matrix,
                 DatasetMapper<PolicyType>& info,
                 PolicyType& policy,
                 const bool fatal,
                 const bool transpose)
```

To implement 

```
template<typename eT, typename PolicyType>
bool Load(const std::string& filename,
                 arma::Mat<eT>& matrix,
                 DatasetMapper<PolicyType>& info,                 
                 const bool fatal,
                 const bool transpose)
```

You can 

```
template<typename eT, typename PolicyType>
inline
bool Load(const std::string& filename,
                 arma::Mat<eT>& matrix,
                 DatasetMapper<PolicyType>& info,                 
                 const bool fatal,
                 const bool transpose)
{
  Load(filename, matrix, info, PolicyType(), fatal, transpose);
}
```

2 : Add a new api in DatasetMapper

`PolicyTypes const& policy() const;`

In this case, we do not need two functions, one function is enough, but the user need to store
the policy in the DatasetMapper before they pass the DatasetMapper into the function.I think this
way is more consistent.